src/main/php/lang/ast/syntax/PHP.class.php

```diff
@@ -860,6 +862,50 @@ public function __construct() {
       return $decl;
     });
 
+    $this->stmt('enum', function($parse, $token) {
+      $name= $parse->scope->resolve($parse->token->value);
+      $parse->forward();
+      $comment= $parse->comment;
+      $parse->comment= null;
+
+      $implements= [];
+      if ('implements' === $parse->token->value) {
+        $parse->forward();
+        do {
+          $implements[]= $parse->scope->resolve($parse->token->value);
+          $parse->forward();
+          if (',' === $parse->token->value) {
+            $parse->forward();
+            continue;
+          } else if ('{' === $parse->token->value) {
+            break;
+          } else {
+            $parse->expecting(', or {', 'interfaces list');
+          }
+        } while (null !== $parse->token->value);
+      }
+      
+      $decl= new EnumDeclaration([], $name, $implements, [], [], $comment, $token->line);
+      $parse->expecting('{', 'enum');
+      $decl->body= $this->typeBody($parse, $decl->name);
```

Comment:
This is not completely correct, as `typeBody()` allows properties, which PHP enums do not. The RFC states this:

> Specifically, the following features of objects are not allowed on enumerations: [...] Enum/Case properties - Properties are a form of state, and enum cases are stateless singletons. Metadata about an enum or case can always be exposed via methods.

Reply:
We'll leave this as-is, the compiler may chose to raise errors about this; or we simply leave it to the PHP runtime.